src/Conversion/KrnlToLLVM/ConvertKrnlToLLVM.cpp

```diff
@@ -220,7 +220,6 @@ void populateAffineAndKrnlToLLVMConversion(RewritePatternSet &patterns,
   arith::populateArithToLLVMConversionPatterns(typeConverter, patterns);
   cf::populateControlFlowToLLVMConversionPatterns(typeConverter, patterns);
 
-  populateReconcileUnrealizedCastsPatterns(patterns);
```

Comment:
In the old llvm, it was used here

/Users/alexe/OM/llvm-project/mlir/lib/Conversion/ReconcileUnrealizedCasts/ReconcileUnrealizedCasts.cpp

Was it replaced with something else there as well? If so, we are lucky. Otherwise, if they changed their approach with regard to unrealized cast, it looks like we will have to somewhat mimic their new approach. Something that we (IBM) should look into.

Thanks for giving it another look, and depending on your finding, we will look into this further.

Much appreciated

Reply:
So this change was in https://github.com/llvm/llvm-project/pull/95700 -- and the suggestion there is to simply remove this and use `-reconcile-unrealized-casts` if needed.

Reply:
`-reconcile-unrealized-casts` is the high level name of the pass, looking at the mods from that PR, it looks like the pass is `mlir::createReconcileUnrealizedCastsPass()` ... so they recommended to remove the missing function, and see if there are any issues; if not, then their suggestion is to call `mlir::createReconcileUnrealizedCastsPass()` pass after the pass that included the removed call.

If you have lit-tests that fails, it is possibly because of the removed pattern. Try to manually run the backend tests (for me: `make -j 12 check-onnx-backend` and see if it runs. If it runs, then it's a matter of correcting the lit tests. I made a tool that can help repair them, we can chat later if that is needed (tool is fixLitTest.py in the utils subdirectories).

Reply:
(if lit tests is an issue, one could also try to add the `-reconcile-unrealized-casts` option to the list of passes executed (which are located in the RUN command at the top of the lit test file).

Reply:
Yes, I added the `reconcile-unrealized-casts` pass to the registration and in some of the lit tests. All the backend tests pass, but several lit tests seem to require quite a few changes. Slowly making my way through them, but would almost certainly need help with some of them.
```
Failed Tests (5):
  Open Neural Network Frontend :: accelerators/nnpa/analysis/dyn-dim-analysis.mlir
  Open Neural Network Frontend :: conversion/krnl_to_llvm/constants_to_file/litte_endian/symbol-postfix.mlir
  Open Neural Network Frontend :: conversion/krnl_to_llvm/entry_point.mlir
  Open Neural Network Frontend :: conversion/krnl_to_llvm/input_verification.mlir
  Open Neural Network Frontend :: onnx/onnx_dim_analysis.mlir
```

Reply:
Any pointers on how to use `fixLitTest.py`?

Reply:
For next time: 
* `fixLitTest.py file.mlir -t` for testing, make a nicer list of failing individual tests
*  `fixLitTest.py file.mlir -t -f func-name` for running an individual test. It create 2 files that can e used to debug, an isolated mlir file of the input, and the output of the compiler. Terminal printout indicates also to run the tests.
* One can manually fix the files, or switch the `-t` (test) for `-r` (repair) and pipe the output in a file. Then the original mlir file and the repaired file can also be compared for further checking.

Note that some of the mlir test files are not fully compliant with the expectations of the format for using `fixLitTest.py`. One may get a file ingestion error if there are missing `\\ ----` file separators. Also, the "fixing" script is not very happy with multiple output operations, so for these, the "repair" function does not fully work. Usually a bit of manual tweaking is sufficient to get the repaired output pass the actual tests.

Reply:
Very nice, thank you!